src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude

```diff
@@ -38,7 +38,4 @@ Microsoft.Extensions.FileSystemGlobbing.dll
 Microsoft.Extensions.Primitives.dll
 Microsoft.Web.XmlTransform.dll
 Newtonsoft.Json.dll
-System.Buffers.dll
-System.Memory.dll
 System.Numerics.Vectors.dll
-System.Runtime.CompilerServices.Unsafe.dll
```

Comment:
Are you removing these due to a build error, or are you proactively removing these?  I see the PR updates the VS SDK build tools package, which could have caused this.

While I expect MSBuild to ship these 3 assemblies, if they don't, or if the assembly loader doesn't find MSBuild's copy, then `msbuild -t:restore` could fail, and we don't have any tests that validates this. The VS pull request will find it, but it's a slow process to fix issues we only catch at VS insertion.

In order to test, you need to build the vsix locally, create a VM and install VS on it, install the vsix, then try to run `msbuild -t:restore` on a solution where packages are downloaded. I'd expect code paths referencing System.Memory and System.Buffers to be exercised.  I don't know when `System.Runtime.CompilerServices.Unsafe.dll` is needed (honestly, it sounds like it might just be compile time attributes and never used at runtime), so a simple test might not catch the code path that needs it, but at least restoring packages will test most of our important code, so there's low risk of regression if it works.

Reply:
These were removed due to a build error, should I still go through this process of verifying restore on a VM?

Reply:
Well, there is risk, although my gut feeling is that it's a low risk. There's increased risk if we ever compile against a higher assembly version than what MSBuild ships, because that will definately cause assembly load failures.

I'm now wondering if it's actually necessary to test a vsix install and msbuild restore. You could try deleting these dlls from `artifacts\NuGet.Build.Tasks.Console\bin\*`, and then use `msbuild -t:restore -p:NuGetRestoreTargets="path\to\artifacts\NuGet.Build.Tasks.Console\bin\...\NuGet.targets"`. Make sure you test with an empty global packages folder and cleaned http-cache, to maximise chance to testing all code paths.  I think this will be equivilent, for the purposes of making sure that msbuild restore doesn't break.